### ISSUE TITLE
Support subproject-only variables and extensions

### ISSUE THREAD

**Issue by uzzu** [uzzu/dotenv-gradle#19]
The plugin refers only to `.env` file in root project directory, and could not create environement variable extension which only to use in subproject.
So organization of environment variable names makes complicated for a bit. (e.g. Appends prefix to name.)

This feature reduces complexity of environment variable names.

```
(Root).
├── .env                 # Adds extension to all subprojects.
├── .env.template        # Adds extension to all subprojects.
├── build.gradle.kts
├── settings.gradle.kts
├── foo
│   ├── .env             # Adds extension only to subproject ':foo'.
│   ├── .env.template    # Adds extension only to subproject ':foo'.
│   ├── build.gradle.kts
│   └── src
└── bar
    ├── .env             # Adds extension only to subproject ':bar'.
    ├── .env.template    # Adds extension only to subproject ':bar'.
    ├── build.gradle.kts
    └── src
```

**Comment by uzzu** [comment]
Supporting entire directory hierarchy is out of scope for this issue, as shown below.

```
(Root).
├── .env                 # Adds extension to all subprojects.
├── .env.template        # Adds extension to all subprojects.
├── build.gradle.kts
├── settings.gradle.kts
└── subprojects
    ├── .env             # [!] Does not refer.
    ├── .env.template    # [!] Does not refer.
    ├── foo
    │   ├── .env             # Adds extension only to subproject ':subprojects:foo'.
    │   ├── .env.template    # Adds extension only to subproject ':subprojects:foo'.
    │   ├── build.gradle.kts
    │   └── src
    └── bar
        ├── .env             # Adds extension only to subproject ':subprojects:bar'.
        ├── .env.template    # Adds extension only to subproject ':subprojects:bar'.
        ├── build.gradle.kts
        └── src
```

Because in Gradle, the directory hierarchy only means default name of subproject. (so it can be changed)

**Comment by abcfy2** [comment]
How is this issue going now? I need this feature. Thanks.

I want to put `.env` file into subproject directory. But it's not working in current version `1.2.0`.

**Comment by uzzu** [comment]
@abcfy2

Thanks for the comments.
Sorry but I had to suspended development.

If you put the .env file in the root project directory, it will be added dotenv variables in the subproject extension. It's currently workaround. But you mean you want to put only into subproject directory, right?

**Comment by abcfy2** [comment]
Yes. If `.env` is in subproject root directory, it should be an subproject only variable.

**Comment by uzzu** [comment]
@abcfy2 

Released in 2.0.0. 🎉 
If you have any problems using it, please create an issue and give us feedback.
